crates/ruff_linter/src/source_kind.rs

```diff
@@ -97,14 +99,31 @@ impl SourceKind {
         match (self, other) {
             (SourceKind::Python(src), SourceKind::Python(dst)) => {
                 let text_diff = TextDiff::from_lines(src, dst);
-                let mut unified_diff = text_diff.unified_diff();
-
-                if let Some(path) = path {
-                    unified_diff.header(&fs::relativize_path(path), &fs::relativize_path(path));
+                let unified_diff = text_diff.unified_diff();
+
+                let relativized_path = if let Some(path) = path {
+                    fs::relativize_path(path)
+                } else {
+                    String::new()
+                };
+
+                // header
+                writeln!(writer, "+++ {}", relativized_path.green())?;
+                writeln!(writer, "--- {}", relativized_path.red())?;
+
+                // individual lines
+                for hunk in unified_diff.iter_hunks() {
+                    let hunk_str = hunk.to_string();
+                    for line in hunk_str.lines() {
+                        let colored_line = match line.chars().next() {
+                            Some('+') => line.green(),
+                            Some('-') => line.red(),
+                            _ => line.normal(),
+                        };
+                        writeln!(writer, "{colored_line}")?;
+                    }
                 }
```

Comment:
Nit: We can iterate over the changes in the `hunk` and use the hunk's display to avoid matching on the first character. 
```rust
								// Individual hunks (section of changes)
                for hunk in unified_diff.iter_hunks() {
                    writeln!(writer, "{}", hunk.header())?;

                    // individual lines
                    for change in hunk.iter_changes() {
                        match change.tag() {
                            ChangeTag::Equal => write!(writer, " {}", change.value())?,
                            ChangeTag::Delete => write!(writer, "{}{}", "-".red(), change.value())?,
                            ChangeTag::Insert => {
                                write!(writer, "{}{}", "+".green(), change.value())?
                            }
                        }
                    }
                }
```

Reply:
Oh this only colors the + and -  compared to also doing `change.value().green()` and `change.value().red()` for the Delete and Insert respectively, or is only coloring the symbols the intended behavior.